sklearn/decomposition/tests/test_pca.py

```diff
@@ -248,21 +248,27 @@ def test_pca_inverse(svd_solver, whiten):
             "auto",
             -1,
             (
-                r"n_components={}L? must be between {}L? and "
-                r"min\(n_samples, n_features\)={}L? with "
-                r"svd_solver=\'{}\'"
+                r"The 'n_components' parameter of PCA must be an int "
+                r"in the range \[0, inf\), a float in the "
+                r"range \(0, 1\), a str among {'mle'} or None. Got -1 instead."
             ),
         ),
         (
             "auto",
             3,
             (
-                r"n_components={}L? must be between {}L? and "
-                r"min\(n_samples, n_features\)={}L? with "
-                r"svd_solver=\'{}\'"
+                r"n_components=3 must be between 0 and min\(n_samples, "
+                r"n_features\)=2 with svd_solver='full'"
```

Comment:
why this change ?

Reply:
Clarity, easy to debug and maintain. Explicitly know the message to expect for that case. Open to reverting it if it's not in line. Also, it's the only case that'll be formatted if reverted.

Reply:
Valid point, let's keep your modification